src/server/saved_objects/client/lib/create_id_query.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Finds a document by either its v5 or v6 format
+ *
+ * @param type The documents type
+ * @param id The documents id or legacy id
+**/
+export function createIdQuery({ type, id }) {
+  return {
+    version: true,
+    size: 1,
+    query: {
+      bool: {
+        should: [
+          // v5 document
```

Comment:
@epixa and I discussed. There are a few changes we're going to make to address this.

First, instead of moving the existing `_id` to `legacyId` and auto-generating `_id`, we're going to set to it something we can expect: `${_type}:${_id}`.

Secondly, we're going to begin including a `saved_object_version` in the exports, beginning with version 2. This will allow us to assume all existing imports are version 1 and should use the concatenated type and ID as the type.